src/plugins/intel_cpu/src/nodes/kernels/jit_base.hpp

```diff
@@ -0,0 +1,199 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <string>
+#include <vector>
+#include <unordered_map>
+#include <math.h>
+#include <dnnl_extension_utils.h>
+#include <cpu/x64/jit_generator.hpp>
+#include <cpu/x64/cpu_isa_traits.hpp>
+#include "registers_pool.hpp"
+#include "stack_allocator.hpp"
+#include "emitters/jit_emitter.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+using namespace dnnl::impl;
+using namespace dnnl::impl::cpu;
+
+class jit_base : public x64::jit_generator {
+public:
+    static constexpr size_t xmm_len = 16;
+    static constexpr size_t ymm_len = 32;
+    static constexpr size_t zmm_len = 64;
+
+    explicit jit_base(x64::cpu_isa_t max_cpu_isa)
+        : x64::jit_generator{nullptr, 256 * 1024, true, max_cpu_isa}
+        , max_cpu_isa_{max_cpu_isa} {}
+    virtual ~jit_base() = default;
+
+    bool is_valid_isa(const x64::cpu_isa_t isa) const {
+        return x64::is_subset(isa, max_cpu_isa_) && x64::mayiuse(isa);
+    }
+
+    using x64::jit_generator::push;
+    using x64::jit_generator::pop;
+
+    void push(const Xbyak::Xmm &xmm) {
+        if (xmm.isXMM()) {
+            sub(rsp, xmm_len);
+            uni_vmovdqu(ptr[rsp], xmm);
+        } else if (xmm.isYMM()) {
+            sub(rsp, ymm_len);
+            uni_vmovdqu(ptr[rsp], Xbyak::Ymm{xmm.getIdx()});
+        } else if (xmm.isZMM()) {
+            sub(rsp, zmm_len);
+            uni_vmovdqu(ptr[rsp], Xbyak::Zmm{xmm.getIdx()});
+        }
+    }
+
+    void push(const std::vector<Xbyak::Xmm> &xmms) {
+        sub(rsp, get_regs_size(xmms));
+        size_t offset = 0;
+        for (const auto& xmm : xmms) {
+            if (xmm.isXMM()) {
+                uni_vmovdqu(ptr[rsp + offset], xmm);
+                offset += xmm_len;
+            } else if (xmm.isYMM()) {
+                uni_vmovdqu(ptr[rsp + offset], Xbyak::Ymm{xmm.getIdx()});
+                offset += ymm_len;
+            } else if (xmm.isZMM()) {
+                uni_vmovdqu(ptr[rsp + offset], Xbyak::Zmm{xmm.getIdx()});
+                offset += zmm_len;
+            }
+        }
+    }
+
+    void pop(const Xbyak::Xmm &xmm) {
+        if (xmm.isXMM()) {
+            uni_vmovdqu(xmm, ptr[rsp]);
+            add(rsp, xmm_len);
+        } else if (xmm.isYMM()) {
+            uni_vmovdqu(Xbyak::Ymm{xmm.getIdx()}, ptr[rsp]);
+            add(rsp, ymm_len);
+        } else if (xmm.isZMM()) {
+            uni_vmovdqu(Xbyak::Zmm{xmm.getIdx()}, ptr[rsp]);
+            add(rsp, zmm_len);
+        }
+    }
+
+    void pop(const std::vector<Xbyak::Xmm> &xmms) {
+        size_t offset = 0;
+        for (const auto& xmm : xmms) {
+            if (xmm.isXMM()) {
+                uni_vmovdqu(ptr[rsp + offset], xmm);
```

Comment:
These should be opposite: from memory address to xmm (now this code block is equal to the similar in push() function)

Reply:
No, it is done properly.
Please, check `preamble` and `postamble`.
In both cases `rsp` refers to the same address. Only difference is that address inscreased and decreased before (in `push`) or after (in `pop`)

Reply:
@avoskoboinyk-lohika fixed